### ISSUE TITLE
Flexible array extension in CC65

### ISSUE THREAD

**Issue by acqn** [cc65/cc65#1126]
https://cc65.github.io/doc/cc65.html#s5:

> cc65 implements flexible array struct members as defined in the C99 ISO standard. As an extension, these fields may be initialized. There are several exceptions, however (which is probably the reason why the standard does not define this feature, because it is highly unorthogonal). Flexible array members cannot be initialized ...
> 
>     ... when defining an array of structs with flexible members.
>     ... if such a struct is a member field of another struct which is not the last field.
>     ... if the struct which contains a flexible array member is declared as register, and the size and compiler settings do allow the compiler actually to place the struct into the register bank in the zero page.
> 
> Please note that -- as defined in the ISO C standard -- the sizeof operator returns the struct size with the flexible array member having size zero, even if it is initialized. 

So how is a flexible array struct member initialized? And how is it supposed to be used after that?

**Comment by jmr** [comment]
You allocate extra space with `malloc` and then initialize it like any other array.

See https://en.cppreference.com/w/c/language/struct and https://en.cppreference.com/w/c/language/array for examples.

**Comment by acqn** [comment]
> 
> 
> You allocate extra space with `malloc` and then initialize it like any other array.
> 
> See https://en.cppreference.com/w/c/language/struct and https://en.cppreference.com/w/c/language/array for examples.

Yeah, that's how it is supposed to be used in ISO C. But the question here is about the "extension" by CC65:
> As an extension, these fields may be initialized.

**Comment by jmr** [comment]
```
struct {
    int x;
    int y[];
} s = { 1, { 2, 3 } }
```

seems to work. gcc has the same extension.

The braces around `2, 3` are required, which is a bit inconsistent with how things usually work, but that looks like a separate bug.

**Comment by acqn** [comment]
The braces issue will be fixed with the solution of Issue #1094. The only thing that is missing is to check if a stuct containing a flexible array member is used as the element type of an array.

**Comment by acqn** [comment]
While we might be able to provide some extension in the future, we should at least give diagnostics on the examples for now.
